types/types.hh

```diff
@@ -962,6 +962,16 @@ utils::chunked_vector<managed_bytes_opt> partially_deserialize_listlike(View in)
 template <FragmentedView View>
 std::vector<std::pair<managed_bytes, managed_bytes>> partially_deserialize_map(View in);
 
+
+// Given a serialized tuple value, reads the nth element and returns it.
+// Returns std::nullopt when there's no element with such index.
+// The second std::nullopt signifies whether the value is NULL or not.
+// make_optional(make_optional(some bytes...)) - a non-null value with these bytes
+// make_optional(std::nullopt) - a null value
+// std::nullopt - no element with this index
```

Comment:
I tried to make `read_nth_tuple_element` a generic function for getting a field of any tuple, so it made sense to notify the caller when any of those cases happen.

In UDTs the value is `null` in both cases (`null field`, `field not present`), but in normal tuples there is a difference. Maybe in the future we would like to be able to access any element of a tuple (`tuple_val[0]`/`tuple_val.0`). Now there's a difference between an existing `null` value and reading an out-of-bounds tuple element. I want this function to be reusable for getting tuple elements, not only UDTs.

I could add a separate function for getting the nth element of a UDT value, but I'm not sure if it's worth it.
It's important to know that for UDTs an out-of-bounds field is considered to have a NULL value, and people who look at/write the code which handles serialized UDTs should be aware of this, which can be forced by making them handle both cases explicitly.

Reply:
Doesn't your last paragraph actually "prove" that we need a read_udt_element() function that knows that an out of bounds field is considered NULL and not something else (not an error, etc.) - i.e., that this logic needs to be written once in the function instead of every caller needing to remember how to handle the two levels of nullopt separately?

I personally think (but don't insist if you think otherwise - you're more an expert in this than me) that if the UDT indexing feature has UDT-specific rules, these rules should be written in a separate UDT-specific function instead of all callers remembering that a UDT is really a tuple with special rules. Moreover, where can the function you added, `read_nth_tuple_element` be used in the future on real tuples? If I understand correctly, CQL doesn't have a syntax to index tuples. Or were you thinking of may adding such a syntax? In that case, wouldn't an out-of-bound index also want to return a null and not anything else?

Reply:
Ok, added `read_nth_user_type_field`